src/com/ableton/VirtualEnv.groovy

```diff
@@ -15,12 +15,59 @@ class VirtualEnv implements Serializable {
    * Destination directory for the virtualenv. This value is set during construction of
    * the object, and is under the system temporary directory.
    */
-  String destDir
+  String destDir = null
+  /**
+   * Series of commands needed to install and activate a version of Python inside the
+   * current shell.
+   */
+  String envSetup = null
+
+  protected String activateSubDir = null
+
+  /**
+   * Create a virtualenv using a specific version of Python, installed via pyenv. pyenv
+   * should be installed on the node, but the actual setup of any required environment
+   * variables (e.g. PYENV_ROOT and PATH) will be done inside this function.
+   *
+   * @param script Script context.
+   *               <strong>Required value, may not be null!</strong>
+   * @param python Python version, as given by pyenv versions --list.
+   * @param pyenvRoot Path to the installation of pyenv.
+   * @return New instance of VirtualEnv object.
+   */
+  static VirtualEnv create(Object script, String python, String pyenvRoot) {
+    if (!script.isUnix()) {
+      script.error 'This method is not supported on Windows'
+    }
 
-  protected String activateSubDir
+    VirtualEnv venv = new VirtualEnv(script, python)
+    assert pyenvRoot
+
+    if (!script.fileExists(pyenvRoot)) {
+      script.error "pyenv root path '${pyenvRoot}' does not exist"
+    }
+
+    venv.envSetup = """
+      export PYENV_ROOT=${pyenvRoot}
+      export PATH=\$PYENV_ROOT/bin:\$PATH
```

Comment:
Why isn't `PYENV_ROOT` set on the node when installing `pyenv`? And why doesn't installing `pyenv` add it to the `PATH`?

Reply:
I'll jump in here for @rco-ableton -- it's because there isn't really a concept of a global environment on Unix, as there is on Windows.

Reply:
I didn't realize that each time we were doing PATH manipulation in cidre (for Python, rust, and CMake), it was always on Windows. 
However, we know and control the user that is running the jobs (`jenkins`), so we could set `PYENV_ROOT` and `PATH` for this user, right?

Reply:
We tried this, but it doesn't work—from memory I think it was something like Jenkins was erasing the user's environment and not picking up the `PYENV_ROOT` etc.

Reply:
Then the current solution is fine I guess.